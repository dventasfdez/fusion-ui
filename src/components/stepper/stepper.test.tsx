import React from "react";
import Stepper, { Step } from "./stepper";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

const StepperTest = (props: any) => (
  <Stepper {...props}>
    <Step name="Step Name" detail="optional" status="success" data-testid="step" />
    <Step name="Step Name" detail="optional" status="pending" />
    <Step name="Step Name" detail="optional" status="disabled" />
    <Step name="Step Name" detail="optional" status="active" />
    <Step name="Step Name" detail="optional" status="error" />
    <Step name="Step Name" detail="optional" status="default" numbered />
    {props.nonValid && <div>Non valid</div>}
  </Stepper>
);

it("Render Stepper and match snapshot", () => {
  const component = renderer.create(<StepperTest data-testid="stepper" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("render stepper vertical", () => {
  const component = render(<StepperTest vertical />);
  expect(component).toBeDefined();
});

it("render with mobile", () => {
  // eslint-disable-next-line no-global-assign
  window = Object.assign(window, { innerWidth: 600 });
  const component = render(<StepperTest />);
  expect(component).toBeDefined();
});

it("render with fake onClick", () => {
  // eslint-disable-next-line no-global-assign
  window = Object.assign(window, { innerWidth: 1024 });
  const onClick = "string";
  const component = render(<StepperTest onClickStep={onClick} />);
  const step = component.getByTestId("step-item-0");
  fireEvent.click(step);
  expect(onClick).toBe("string");
});

it("render with onClick", () => {
  const onClick = jest.fn();
  // eslint-disable-next-line no-global-assign
  window = Object.assign(window, { innerWidth: 1024 });
  const component = render(<StepperTest onClickStep={onClick} data-testid="stepper" />);
  const step = component.getByTestId("step-item-0");
  fireEvent.click(step);
  expect(onClick).toHaveBeenCalled();
});

it("render with non valid children", () => {
  const component = render(<StepperTest nonValid />);
  expect(component).toBeDefined();
});
