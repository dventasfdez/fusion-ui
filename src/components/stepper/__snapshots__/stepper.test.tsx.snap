// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Stepper and match snapshot 1`] = `
<div
  className="stepper "
  data-testid="stepper"
>
  <div
    className="stepper-step-container success"
    data-testid="step-item-0"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span
          className="material-icons"
        >
          check
        </span>
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container pending"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container disabled"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container active"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container error"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span
          className="material-icons"
        >
          error_outline
        </span>
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container default"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container success"
    data-testid="step-item-undefined"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span
          className="material-icons"
        >
          check
        </span>
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container pending"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container disabled"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container active"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span />
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container error"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span
          className="material-icons"
        >
          error_outline
        </span>
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
  <div
    className="stepper-step-container default"
    onClick={[Function]}
  >
    <div
      className="stepper-item-content"
    >
      <div
        className="stepper-step"
      >
        <span
          className="step-number"
        >
          NaN
        </span>
      </div>
    </div>
    <div
      className="stepper-text-content"
    >
      <p
        className="stepper-title"
      >
        Step Name
      </p>
      <p
        className="stepper-subtitle"
      >
        optional
      </p>
    </div>
  </div>
</div>
`;
